packages/core/src/WorkflowExecutionMetadata.ts

```diff
@@ -17,16 +29,30 @@ export function setWorkflowExecutionMetadata(
 	) {
 		return;
 	}
-	executionData.resultData.metadata[String(key).slice(0, 50)] = String(value).slice(0, 255);
+	if (typeof key !== 'string') {
+		throw new WorkflowMetadataValidationError('key', key);
```

Comment:
nit: I think it's better we call it `ExecutionMetadataValidationError` since it relates tot he execution rather than the workflow itself